### PR TITLE
refactor(api,shared-data,robot-server):  ot3 deck

### PR DESCRIPTION
**Pull request by sfoster1**
The OT-3 will have a different shaped deck than the OT-2. There is enough OT-3 hardware that we can determine what that deck looks like.

This PR which was actually done by @Laura-Danielle adds the definition and uses it throughout the backend by centralizing the deck type based on the ot3 feature flag.

I'm unsure whether this should be merged before 5.0 or not. The changes should be well tested.

@Opentrons/app-and-uis is this going to break anything in OT-2 protocols if we don't add it to the frontends?

### CODE REVIEW

**Review by Laura-Danielle — APPROVED** [review]

File: shared-data/deck/schemas/2.json [hunk 1]
```diff
@@ -112,7 +112,7 @@
         "model": {
           "description": "Model of the robot (currently only OT-2 Standard is supported)",
```
Comment (b-cooper) [inline shared-data/deck/schemas/2.json]: Looks like this description needs to change too.
```suggestion
          "description": "Model of the robot",
```

**Review by b-cooper — APPROVED** [review]
I don't foresee any problems on the frontend with this merging before or after 5.0.0.  Every time we reference the deck on the FE we're using the deck's `load_name`

That said, glancing at the definition, it looks like a fair amount of it is copy pasta from the ot2 standard deck and very likely isn't valid for the ot3 deck. As long as we're aware of this and document what parts of the definition still need to be constructed. I'm good with merging this.

Quick question:
In terms of slot naming, I think we should consider numbering the slots Left-to-Right/Back-to-Front , instead of Left-to-Right/Front-to-Back.  This lines up more with expectations of ordering for labware in general. E.g. custom decks that have been made for PRL follow this convention with slot 1 in the back left. This would also ideally be the origin point of the deck and the y-axis would flip to be positive coming towards the front of the robot.

**Comment by sfoster1** [comment]
> Quick question: In terms of slot naming, I think we should consider numbering the slots Left-to-Right/Back-to-Front , instead of Left-to-Right/Front-to-Back. This lines up more with expectations of ordering for labware in general. E.g. custom decks that have been made for PRL follow this convention with slot 1 in the back left. This would also ideally be the origin point of the deck and the y-axis would flip to be positive coming towards the front of the robot.

Uh I guess. We should bring that up with hardware since presumably the slot covers will have embossed numbers on them that would have to match. FOr now let's keep it the same.

**Review by shlokamin — APPROVED** [review]
ditto what @b-cooper said, don't think this affects app & ui 5.0 work

File: api/tests/opentrons/protocol_engine/resources/test_deck_data_provider.py [hunk 2]
```diff
@@ -38,9 +39,10 @@ async def test_get_deck_definition(
 async def test_get_deck_definition_short_trash(
     short_trash_deck_def: DeckDefinitionV2,
     subject: DeckDataProvider,
+    short_trash_flag: AsyncGenerator[None, None],
```
Comment (mcous) [inline api/tests/opentrons/protocol_engine/resources/test_deck_data_provider.py]:
```suggestion
    short_trash_flag: None,
```

**Review by mcous — APPROVED** [review]

**Comment by Laura-Danielle** [comment]
> I don't foresee any problems on the frontend with this merging before or after 5.0.0. Every time we reference the deck on the FE we're using the deck's `load_name`
> 
> That said, glancing at the definition, it looks like a fair amount of it is copy pasta from the ot2 standard deck and very likely isn't valid for the ot3 deck. As long as we're aware of this and document what parts of the definition still need to be constructed. I'm good with merging this.
> 
> Quick question: In terms of slot naming, I think we should consider numbering the slots Left-to-Right/Back-to-Front , instead of Left-to-Right/Front-to-Back. This lines up more with expectations of ordering for labware in general. E.g. custom decks that have been made for PRL follow this convention with slot 1 in the back left. This would also ideally be the origin point of the deck and the y-axis would flip to be positive coming towards the front of the robot.

Yep we're definitely aware that a lot needs to change. This is mostly a placeholder for us to continue working on higher priority stuff such as fine-tuning motion control. We've got stuff ticketed for making a new deck definition (and most likely a new schema version as well).

**Pull request merged**